tidy3d/components/autograd/derivative_utils.py

```diff
@@ -16,6 +17,62 @@
 PermittivityData = dict[str, ScalarFieldDataArray]
 
 
+class DerivativeSurfaceMesh(Tidy3dBaseModel):
+    """Stores information about the surfaces of an object to be used for derivative calculation.
+
+    user guide: this class is used to construct derivatives with respect to surface elements
+    given some forward and adjoint fields stored in a ``DerivativeInfo`` class. To use it,
+    you must specify the central locations of all the surface elements in ``centers``,
+    along with the area of each element in ``areas``. Then you need to specify three orthogonal
+    vectors (``normals``, ``perps1`` and ``perps2``). The derivative will be computed with
+    respect to a change in material along the ``normals`` direction. It is important to note
+    that the sign of these basis vectors is irrelevant since we end up multiplying the forward
+    and adjoint fields together in these bases.
```

Comment:
sure, sounds good. I think I also ended up writing that last paragraph because after re-reading I realized it needed more clarification, so makes sense to put there.